src/silx/app/view/CustomPlotSelectionWindow.py

```diff
@@ -122,15 +128,18 @@ def __init__(self, plot, parent=None):
 
         self.addUrl(silx.io.url.DataUrl())
 
-        self.rowsAboutToBeRemoved.connect(self._rowAboutToBeRemoved)
+        self.rowsAboutToBeRemoved.connect(self._rowsAboutToBeRemoved)
 
-    def getXParent(self):
+    def getXParent(self) -> qt.QStandardItem:
+        """Return the parent item for the X dataset"""
         return self._xParent
 
-    def getYParent(self):
+    def getYParent(self) -> qt.QStandardItem:
+        """Return the parent item for the Y datasets"""
         return self._yParent
 
-    def _setXFile(self, url=None, curve=None):
+    def _setXFile(self, url: silx.io.url.DataUrl | None = None, curve: None = None):
```

Comment:
The `curve` argument is always `None`, better to remove it (this needs to change its usage in the function body as well):
```suggestion
    def _setXFile(self, url: silx.io.url.DataUrl | None = None):
```